pkg/logging/logger.go

```diff
@@ -21,11 +21,33 @@ func GetLogger(prefix string) *logrus.Entry {
 	return log.WithField("prefix", prefix)
 }
 
+// AddField add a field to an existing logrus.Entry
+func AddField(e *logrus.Entry, name string, value interface{}) *logrus.Entry {
+	l := log.WithField(name, value)
+	for k, v := range e.Data {
+		l = l.WithField(k, v)
+	}
+	return l
+}
+
+// AddFields adds multiple fields to an existing logrus.Entry
+func AddFields(e *logrus.Entry, fields map[string]interface{}) *logrus.Entry {
+	l := e
+	for k, v := range fields {
+		l = AddField(l, k, v)
```

Comment:
My understanding of `WithFields` is what it will create a new logger and add the requested fields. The intention here is to add fields to an existing logger, already associated with other fields. So basically what it does is calling `log.WithField`, but then is also iterates on `Data` on the existing logger and adds those fields to the newly created object. As far as I could tell when I put this up, `logrus` would not support doing this.